velox/exec/Window.cpp

```diff
@@ -147,14 +281,70 @@ void Window::createWindowFunctions(
         windowNodeFunction.functionCall->type(),
         operatorCtx_->pool()));
 
-    checkDefaultWindowFrame(windowNodeFunction);
-
-    windowFrames_.push_back(
+    auto checkFrameType = [&](const std::string errorMessage) -> void {
+      if (windowNodeFunction.frame.type ==
+          core::WindowNode::WindowType::kRange) {
+        VELOX_FAIL(errorMessage);
+      }
+    };
+    auto updateKBasedFrameBounds = [&](const core::TypedExprPtr& frame,
+                                       WindowFrame& windowFrame,
+                                       bool isStartFrame) -> void {
+      if (auto constant = constantArg(frame)) {
+        VELOX_NYI(
+            "k must be a column, constant INTEGER offsets are not supported");
+      } else {
+        FlatVectorPtr<int64_t> offsets =
+            std::dynamic_pointer_cast<FlatVector<int64_t>>(
+                BaseVector::create(BIGINT(), 0, operatorCtx_->pool()));
+        if (isStartFrame) {
+          windowFrame.startFrameArg =
+              FrameArg({false, fieldArgToChannel(frame).value(), offsets});
```

Comment:
If the first argument is always false then why is it needed ?

Reply:
In the other PR for constant k values, it is true, so this argument is necessary in FrameArg.